pkg/apis/planetscale/v2/vitessbackupstorage_types.go

```diff
@@ -74,6 +74,9 @@ type VitessBackupLocation struct {
 	// +kubebuilder:validation:MaxLength=25
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([a-z0-9]*[a-z0-9])?$
 	Name string `json:"name,omitempty"`
+	// SubPath gives the subpath in the volume to mount to the backups target.
+	// Only used for Volume-backed backup storage, ignored otherwise.
+	SubPath     string `json:"subPath,omitempty"`
```

Comment:
Since this only applies to Volume-based backup locations, how about we call it `VolumeSubPath` to help signal that?

I really should have made `Volume` a struct with `Source *corev1.VolumeSource` inside it, so we could now add `SubPath`, but unfortunately I didn't think of it.